sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubProducerClient.cs

```diff
@@ -330,7 +516,25 @@ protected EventHubProducer()
         ///
         /// <returns>A task to be resolved on when the operation has completed.</returns>
         ///
-        public virtual Task CloseAsync(CancellationToken cancellationToken = default) => InnerProducer.CloseAsync(cancellationToken);
+        public virtual async Task CloseAsync(CancellationToken cancellationToken = default)
+        {
+           _closed = true;
+
+           try
+           {
+                await InnerProducer.CloseAsync(cancellationToken).ConfigureAwait(false);
+
+                if (OwnsConnection)
+                {
+                    await Connection.CloseAsync().ConfigureAwait(false);
```

Comment:
Cancellation token is deliberately not being passed, as the producer is closed and owns the connection;  we don't want to leave things in an inconsistent state.